fastapi/dependencies/utils.py

```diff
@@ -571,12 +580,22 @@ async def solve_dependencies(
     values.update(header_values)
     values.update(cookie_values)
     errors += path_errors + query_errors + header_errors + cookie_errors
+
+    if check_unknown:
+        dependant_names = {param.name for param in dependant.query_params}
+        left_query_params.difference_update(dependant_names)
+        if outer_level:
```

Comment:
Merge this condition to the upper one. Avoid nested conditions if you do not have to.